bin/.gitignore

```diff
@@ -1,2 +1,2 @@
 cache
-
+flutter.lock
```

Comment:
On Linux, the bash script itself is used as the lock file [1]. This doesn't work on Windows because a file that's open for write cannot be opened for read: Trying to start a `flutter.bat` instance while another `flutter.bat` is running and holding a write lock on `flutter.bat` results in an (uncatchable) OS-level error.

Maybe you are thinking about the `lockfile` we use on all platforms in dart land [2]. Re-using that one would change the semantics slightly because now dart land and the batch script would share the same lock. That's not necessary, though, because the batch script's download of the dart-sdk in one instance of flutter_tools and dart land's download of engine artifacts in another instance of flutter_tools can happen in parallel without problems.

[1] https://github.com/flutter/flutter/blob/master/bin/flutter#L79
[2] https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/cache.dart#L57